llvm/test/ThinLTO/X86/memprof-icp-recursive.ll

```diff
@@ -54,7 +54,40 @@
 ; RUN:  -o %t.out 2>&1 | FileCheck %s --check-prefix=STATS \
 ; RUN:  --check-prefix=REMARKS
 
-; RUN: llvm-dis %t.out.2.4.opt.bc -o - | FileCheck %s --check-prefix=IR
+; RUN: llvm-dis %t.out.2.4.opt.bc -o - | FileCheck %s --check-prefixes=IR,IR-INLINE
+
+;; Next, add a threshold to prevent inlining of the promoted calls which have
+;; count 2 (the default threshold of 2 means they are inlinable by default).
+; RUN: llvm-lto2 run %t/main.o %t/foo.o -enable-memprof-context-disambiguation \
+; RUN:	-memprof-icp-noinline-threshold=3 \
```

Comment:
The synthetic values are using 1, so a threshold of 2 should be sufficient with a strict < check?

Reply:
I guess this is a non-default value that is being tested?

Reply:
This is not a synthetic VP metadata. It has a count of 2, so by default it does not add noinline, as desired. I added the flag in here just to test that it works as intended.

Reply:
got it, thanks.